search-commons/src/main/java/no/unit/nva/search2/ResourceQuery.java

```diff
@@ -45,14 +45,13 @@ private ResourceQuery() {
 
     @SuppressWarnings("PMD.NullAssignment")
     @NotNull
-    public PagedSearchResourceDto toPagedSearchResponseDto(@NotNull OpenSearchSwsResponse response) {
+    private PagedSearchResourceDto toPagedSearchResponseDto(@NotNull OpenSearchSwsResponse response) {
 
-        var offset = getQueryOffset();
-        var url = gatewayUri.toString().split("\\?")[0];
-        var requestParameter = toGateWayRequestParameter();
-        var id = createUriOffsetRef(url, requestParameter, offset);
-
-        var hasMoreResults = offset < response.getTotalSize();
+        final var offset = getQueryOffset();
+        final var url = gatewayUri.toString().split("\\?")[0];
+        final var requestParameter = toGateWayRequestParameter();
+        final var id = createUriOffsetRef(url, requestParameter, offset);
+        final var hasMoreResults = offset < response.getTotalSize();
         var nextResults
```

Comment:
I tried to extract to functions, it didn't not improve readability, without doing unnecessary calls